libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -740,8 +761,14 @@ void IRGenerator::generateConstructors(ContractDefinition const& _contract)
 			)");
 			vector<string> params;
 			if (contract->constructor())
+			{
+				t("sourceLocationComment", sourceLocationComment(*contract->constructor(), m_context));
```

Comment:
Would
```suggestion
				t("sourceLocationComment", sourceLocationComment(*(contract->constructor() ? contract->constructor() : contract), m_context));
```
work (and put it outside the if)?

Reply:
Sure, my intention was to use the existing if check instead of checking the same thing twice...
If you think this is more readable, I can change it..